src/parser/Parser.cpp

```diff
@@ -1,122 +1,115 @@
 #include <Parser.h>
 
-Parser::Parser(Lexer lexer)
-    : m_lexer(std::move(lexer))
-    , m_currentToken(m_lexer.getToken())
+Parser::Parser(const Lexer lexer)
+    : m_lexer(lexer)
 {
+    m_currenttoken = m_lexer.getToken();
+}
 
+Token Parser::eattoken(TokenType t) {
+    if (m_currenttoken.type != t) {
+        puts("Unexpected token type.");
+        exit(1);
+    }
+    m_currenttoken = m_lexer.getToken();
 }
 
-void Parser::buildTree() {
-    parseStatement();
+std::unique_ptr<AST::Stmt> Parser::parse() {
+    return parseStmt();
 }
 
-std::unique_ptr<AST::Stmt> Parser::parseStatement() {
-    std::unique_ptr<AST::Stmt> statement;
-    Token identifier;
-    switch(m_currentToken.type) {
-        case TokenType::Def:
-            statement = parseFunction();
-        case TokenType::Return:
-            statement = parseReturn();
-        case TokenType::Identifier:
-            identifier = m_currentToken;
-            getNextToken();
-            switch(m_currentToken.type) {
-                case TokenType::LeftParen:
-                    getNextToken();
-            }
-            break;
-
-        // todo: other statements
-
-        default:
-            statement = parseExpression();
+std::unique_ptr<AST::Stmt> Parser::parseStmt(){
```

Comment:
I'm not sure if the comment I left earlier will show up but I believe we still need to handle the `EOL`/`END` case here, otherwise you'll crash as soon as you reach the end of input.

Reply:
we'll handle that at a high level than this, i.e. call parse(), then eat a semicolon, then call parse, etc.